indico/modules/events/papers/models/call_for_papers.py

```diff
@@ -69,3 +71,41 @@ def set_reviewing_state(self, reviewing_type, enable):
             paper_reviewing_settings.set(self.event, 'content_reviewing_enabled', enable)
         elif reviewing_type == 'layout':
             paper_reviewing_settings.set(self.event, 'layout_reviewing_enabled', enable)
+
+    @property
+    def managers(self):
+        return {p.principal for p in self.event.acl_entries
+                if p.has_management_role('paper_manager', explicit=True)}
+
+    @property
+    def judges(self):
+        return {p.principal for p in self.event.acl_entries
+                if p.has_management_role('paper_judge', explicit=True)}
+
+    @property
+    def content_reviewers(self):
+        return {p.principal for p in self.event.acl_entries
+                if p.has_management_role('paper_content_reviewer', explicit=True)}
+
+    @property
+    def layout_reviewers(self):
+        return {p.principal for p in self.event.acl_entries
+                if p.has_management_role('paper_layout_reviewer', explicit=True)}
+
+    @property
+    def acl_entries(self):
+        entries = self.managers | self.judges
+        if self.content_reviewing_enabled:
+            entries = entries | self.content_reviewers
+        if self.layout_reviewing_enabled:
+            entries = entries | self.layout_reviewers
+        return entries
+
+    @property
+    @memoize_request
+    def user_competences(self):
+        user_ids = {user.id for user in self.event.cfp.acl_entries}
+        user_competences = (PaperCompetence.query
+                            .with_parent(self.event)
+                            .filter(PaperCompetence.user_id.in_(user_ids)).all())
```

Comment:
I'd put `.all()` into a new line

Also, usually I keep the `.with_parent(..)` in the `Foo.query` line but no strong opinion, I just see it more as something that's giving you an initial query while everything else after that modifies/executes it